File: typescript/src/points/cartesianpoint.ts
interface CartesianPoint extends Point {
    x: number,
    y: number
}

function newCartesianPoint (x: number, y: number): CartesianPoint {
    return {
        x,
        y,

        getX: function () : number {
            return x;
        },
        getY: function () : number {
            return y;
        }
    }
}

export { CartesianPoint, newCartesianPoint }

File: java/src/main/java/edu/cmu/cs/cs214/rec02/points/CartesianPoint.java
package edu.cmu.cs.cs214.rec02.points;

public class CartesianPoint implements Point {
    private int x;
    private int y;

    public CartesianPoint (int x, int y) {
        this.x = x;
        this.y = y;
    }
    public int getX() {
        return this.x;
    }
    public int getY() {
        return this.y;
    }
}
